### PR TITLE
led subsystem (WIP)

### LINKED ISSUE #21: Creating/modifying the LED subsystem

### PR DESCRIPTION
**Pull request by m10653**
Closes #21

### CODE REVIEW

File: src/main/java/org/jmhsrobotics/frc2024/subsystems/LED/LEDSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package org.jmhsrobotics.frc2024.subsystems.LED;
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj.simulation.AddressableLEDSim;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class LEDSubsystem extends SubsystemBase {
+	private AddressableLED LED;
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/subsystems/LED/LEDSubsystem.java:10]: rename to lowercase led. Names with capital letters are generally used for Classes and Constants.

File: src/main/java/org/jmhsrobotics/frc2024/subsystems/LED/commands/LEDCommand.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package org.jmhsrobotics.frc2024.subsystems.LED.commands;
+
+import org.jmhsrobotics.frc2024.subsystems.LED.LEDSubsystem;
+
+import edu.wpi.first.wpilibj2.command.Command;
+
+public class LEDCommand extends Command {
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/subsystems/LED/commands/LEDCommand.java:7]: Rename Command to denote that it produces a Rainbow/ or add a parameter to provide a color.

File: src/main/java/org/jmhsrobotics/frc2024/subsystems/LED/LEDSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	public LEDSubsystem() {
+		if (RobotBase.isSimulation()) {
+			initSim();
+		}
+
+		// TODO: find the real port
+		this.LED = new AddressableLED(9);
+
+		// TODO: get the real length
+		this.ledBuffer = new AddressableLEDBuffer(60);
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/subsystems/LED/LEDSubsystem.java:23]: Move Values for Ports and LED length into a constants file/class

File: src/main/java/org/jmhsrobotics/frc2024/subsystems/LED/commands/LEDCommand.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		this.ledSubsystem.startLED();
+
+	}
+
+	@Override
+	public void execute() {
+
+		// create a rainbow effect
+		// this.ledSubsystem.rainBow();
+		this.ledSubsystem.setRGB(255, 0, 0);
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/subsystems/LED/commands/LEDCommand.java:28]: If setting a solid color method only needs to be set once within ```initialize``` If we are doing rainbow this is fine to do within excecated as the databuffer for the leds changes every loop.

**Review by m10653 — COMMENTED** [review]
Recommended changes.

File: src/main/java/org/jmhsrobotics/frc2024/subsystems/LED/LEDSubsystem.java (hunk 3 above)
Comment (Lu-han-wang) [reply src/main/java/org/jmhsrobotics/frc2024/subsystems/LED/LEDSubsystem.java]: I created a new issue to create a global constant file and do the clean up

**Review by Lu-han-wang — APPROVED** [review]

**Pull request merged**